README.md

```diff
@@ -1,3 +1,246 @@
+
+- These build-time arguments don't persist when you launch a container instance
+  from the image (in contrast to the `ENV` directive that sets environment
+  variables that persist once the image is built).
+
+- If you declare a build argument with `ARG`, you can pass
+  `--build-arg ARG_NAME=some_value` when you run your `docker build` command to
+  override whatever the default value of that argument is.
+
+- Example: your image build depends on specific versions of another package you
+  want to install, to make the image reusable, you may want to pass the package
+  version as a build argument so that you can build different versions of the
+  image over time.
+
+- Example: you might want to set different build-time variables if you are
+  building your image for different hosts (e.g. the URL for a proxy server).
+
+- See [documentation on Docker builds][docker-build-args] for more information.
+
+**Recommendation:** where applicable, consider using build-time arguments by
+using `ARG` and `--build-arg` to declare and override build-time arguments.
+
+### Lint your Dockerfile
+
+- There is a tool called [hadolint][] that you can use to lint your Dockerfile.
+
+- The linter will indicate areas where you can improve your Dockerfile and also
+  provide suggestions.
+
+**Recommendation:** lint your Dockerfile to improve code quality.
+
+### Understand when not to use Alpine-based images
+
+- A popular minimal image that is used as a base image in the Docker community
+  is the [alpine][] image, which is based off of the Alpine distribution of
+  Linux.
+
+- This is a very lightweight image that has many use cases, but this is often
+  not a good choice as a base image for data science oriented projects (esp. it
+  is not a great choice for projects with heavy Python dependencies).
+
+- Why?
+
+  - Reason is that most python packages include prebuilt wheels (wheels are a
+    kind of pre-built distribution for python that allow you to avoid the build
+    stage that is normally required with source distributions).
+  - Since Alpine is very stripped down compared to other Linux distributions, it
+    has a different version of the standard C library that is required by most C
+    programs (including Python).
+  - Because of the above, Alpine doesn't support Python wheels, so when you
+    install Python packages on an Alpine based image, you install the source
+    code, which means that all C code in every Python package you use must be
+    compiled.
+  - This also means you need to figure out every system library dependency
+    yourself.
+
+- **Bottom Line:** If you try to install a bunch of Python packages on an Alpine
+  based image, your build will take longer, be more error prone, and your disk
+  footprint will be higher than if you installed those Python packages on an
+  image based on Debian or another Linux distribution with the full standard C
+  library (`glibc`).
```

Comment:
I think we shouldn't be opinionated about `full standard C library`, but can just acknowledge that most python wheels are only available for `glibc`.